packages/front-end/components/Settings/UpgradeModal/index.tsx

```diff
@@ -80,37 +78,46 @@ export default function UpgradeModal({
 
   const { organization, refreshOrganization } = useUser();
 
+  function getLowestAvailablePlan() {
+    // If no commercialFeature is passed in, the lowestAvailablePlan is the plan higher than their effectiveAccountPlan
+    if (!commercialFeature) {
+      if (
+        !effectiveAccountPlan ||
+        ["oss", "starter"].includes(effectiveAccountPlan)
+      ) {
+        return "pro";
+      }
+
+      if (
+        effectiveAccountPlan &&
+        ["pro", "pro_sso"].includes(effectiveAccountPlan)
+      ) {
+        return "enterprise";
+      }
+    } else {
+      return commercialFeatureLowestPlan?.[commercialFeature] || "starter";
```

Comment:
A couple of questions about this line:
Is typescript making sure that `commercialFeature` is always a valid key of `commercialFeatureLowestPlan`?  
There is no "starter" plan modal, should we be defaulting to "pro"?
`starter` plan is only on cloud, while `oss` is for self-serve.  Though I don't think there is much difference in the logic for how we handle the two right now, we should probably default to the correct "free" version.

Reply:
Ok I see below that we just check to see if this equals "enterprise" below. What about renaming this function `showEnterpriseTreatment` or `showEnterpriseVersion` and have it return a boolean?

Reply:
I'm good with that change, I'll make it now.